core/src/main/java/org/apache/carbondata/core/datastore/compression/GzipCompressor.java

```diff
@@ -0,0 +1,132 @@
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.carbondata.core.datastore.compression;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+
+import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream;
+import org.apache.commons.compress.compressors.gzip.GzipCompressorOutputStream;
+
+/**
+ * Codec Class for performing Gzip Compression
+ */
+public class GzipCompressor extends AbstractCompressor {
+
+  @Override public String getName() {
+    return "gzip";
+  }
+
+  /**
+   * This method takes the Byte Array data and Compresses in gzip format
+   *
+   * @param data Data Byte Array passed for compression
+   * @return Compressed Byte Array
+   */
+  private byte[] compressData(byte[] data) {
+    ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+    try {
+      GzipCompressorOutputStream gzipCompressorOutputStream =
+          new GzipCompressorOutputStream(byteArrayOutputStream);
+      try {
+        /**
+         * Below api will write bytes from specified byte array to the gzipCompressorOutputStream
+         * The output stream will compress the given byte array.
+         */
+        gzipCompressorOutputStream.write(data);
+      } catch (IOException e) {
+        throw new RuntimeException("Error during Compression writing step ", e);
+      } finally {
+        gzipCompressorOutputStream.close();
+      }
+    } catch (IOException e) {
+      throw new RuntimeException("Error during Compression step ", e);
+    }
+    return byteArrayOutputStream.toByteArray();
+  }
+
+  /**
+   * This method takes the Byte Array data and Decompresses in gzip format
+   *
+   * @param data   Data Byte Array for Compression
+   * @param offset Start value of Data Byte Array
+   * @param length Size of Byte Array
+   * @return
+   */
+  private byte[] decompressData(byte[] data, int offset, int length) {
+    ByteArrayInputStream byteArrayOutputStream = new ByteArrayInputStream(data, offset, length);
+    ByteArrayOutputStream byteOutputStream = new ByteArrayOutputStream();
+    try {
+      GzipCompressorInputStream gzipCompressorInputStream =
+          new GzipCompressorInputStream(byteArrayOutputStream);
+      byte[] buffer = new byte[1024];
```

Comment:
Instead of fixed 1024, can you observe what is the blocksize (bytes size) gzip operates and use the same value ?

Reply:
Yeah, I can fixed it with double of the data length, so it minimizes the read. I have added that.